### PR TITLE
[Term Entry] Kotlin datetime function: .DateTimePeriod()

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added the datetime.DateTimePeriod() term for Kotlin
- Closes issue #3809 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+Title: '.DateTimePeriod()'
+Description: 'Returns a DateTimePeriod object or a DatePeriod object (if all arguments are set to zero) to indicate the time difference between two dates.'
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:3]:
```suggestion
Description: 'Returns a DateTimePeriod object to indicate the time difference between two dates.'
```

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Tags:
+  - 'Methods'
+  - 'Data Structures'
+  - 'Time'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`DateTimePeriod`** function helps to make a new `DateTimePeriod` object that tells you how much time has passed between two dates (even up to nanoseconds). If nothing is passed into the function, it returns a `DatePeriod` object by default. You can use this function to set how many years, months, days, hours, minutes, seconds, and even nanoseconds are between two dates.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:16]:
```suggestion
The **`.DateTimePeriod`** function helps to make a new `DateTimePeriod` object that tells you how much time has passed between two dates (even up to nanoseconds). If nothing is passed into the function, it returns a `DatePeriod` object by default. You can use this function to set how many years, months, days, hours, minutes, seconds, and even nanoseconds are between two dates.
```

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+```pseudo
+fun DateTimePeriod(
+    years: Int = 0,
+    months: Int = 0,
+    days: Int = 0,
+    hours: Int = 0,
+    minutes: Int = 0,
+    seconds: Int = 0,
+    nanoseconds: Long = 0
+): DateTimePeriod
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:29]: The syntax contains plagiarism, please try to remove it.

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+The function takes in up to seven parameters, and it is recommended to always explicitly name the arguments when constructing the function, like `DateTimePeriod(years = 1, months = 12)`. If all the time components are set to zero, the function will return a `DatePeriod` object.
+
+The function is also streamlined for human readability; for example, `DateTimePeriod(months = 24)` becomes `DateTimePeriod(years = 2)`.
+
+Note that the function will throw an `IllegalArgumentException` if any of the inputs overflows a Long.
+
+## Example
+
+The following example shows how to use the **`DateTimePeriod`** function to add two weeks to a given date.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:40]:
```suggestion
The following example shows how to use the **`.DateTimePeriod`** function to add two weeks to a given date.
```

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  val currDate = "2023-12-25T12:00:00.00+00:00".toInstant()
+
+  // Create a new DateTimePeriod for 2 weeks
+  val twoWeeks = DateTimePeriod(days = 14)
+
+  // add 2 weeks to the current date
+  val newDate = currDate.plus(twoWeeks, TimeZone.UTC)
+
+  println(newDate)
+}
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:55]:
```suggestion
import kotlinx.datetime.DateTimePeriod
fun main() {
  //Create an Instant for 2023-12-25 12:00 UTC zone from an ISO 8601 String
  val currDate = "2023-12-25T12:00:00.00+00:00".toInstant()

  // Create a new DateTimePeriod for 2 weeks
  val twoWeeks = DateTimePeriod(days = 14)

  // add 2 weeks to the current date
  val newDate = currDate.plus(twoWeeks, TimeZone.UTC)

  println(newDate)
}
```
The program is giving errors, please add all the necessary import statements.

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+The output will be similar to the following:
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:58]:
```suggestion
The output is as follows:
```

File: content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    seconds: Int = 0,
+    nanoseconds: Long = 0
+): DateTimePeriod
+```
+
+The function takes in up to seven parameters, and it is recommended to always explicitly name the arguments when constructing the function, like `DateTimePeriod(years = 1, months = 12)`. If all the time components are set to zero, the function will return a `DatePeriod` object.
+
+The function is also streamlined for human readability; for example, `DateTimePeriod(months = 24)` becomes `DateTimePeriod(years = 2)`.
+
+Note that the function will throw an `IllegalArgumentException` if any of the inputs overflows a Long.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/DateTimePeriod/DateTimePeriod.md:36]: Add it as a Note, take reference from this [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md).

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@andersooi Please make these changes and only commit DateTimePeriod.md file, don't commit the rest of the two files.

**Comment by andersooi** [comment]
Hi @avdhoottt thanks for the comments. After I make the changes to the DateTimePeriod.md file, do I just add the file to the staging area, commit and push? Or do I need to uncommit the other 2 files first before doing that (apologies as I am still learning how to use GitHub)?

**Comment by avdhoottt** [comment]
@andersooi Delete the files from the branch, Once files are deleted commit the change and push the branch.

**Comment by avdhoottt** [comment]
@andersooi Don't just delete them directly from Github, remove them, commit the change, and then push the branch.

**Comment by andersooi** [comment]
Thanks @avdhoottt. I am currently in the `docs` working directory.

So I ran the following:
`git rm --cached package.json` and `git rm --cached yarn.lock`, but was met with the error that it did not match any files.

I was thinking of resetting git to the commit SHA of my first commit (the one that I committed all 3 files), run the same commands again, commit and then push.

Just wanted to check if this is alright or if I should be using another workaround.

**Comment by avdhoottt** [comment]
@andersooi can you revert back to the commit where you have not committed those files? or you can create a new Pull Request but remember to commit the necessary file only.

**Comment by andersooi** [comment]
Hi @avdhoottt I shall create a new pull request instead and commit the necessary file only. Will close this pull request.

**Pull request closed without merging**